components/org.wso2.carbon.editor.log.appender/src/main/java/org/wso2/carbon/editor/log/appender/EditorConsoleAppender.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 WSO2, Inc. (http://wso2.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.editor.log.appender;
+
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.Core;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+import org.apache.logging.log4j.core.util.Booleans;
+import org.wso2.carbon.editor.log.appender.internal.CircularBuffer;
+import org.wso2.carbon.editor.log.appender.internal.ConsoleLogEvent;
+
+import java.io.Serializable;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * This appender will be used to capture the logs and later send to clients, if requested via the
+ * logging web service.
+ * This maintains a circular buffer, of some fixed amount (say 100).
+ */
+@Plugin(name = "EditorConsole", category = Core.CATEGORY_NAME, elementType = Appender.ELEMENT_TYPE, printObject = true)
+public final class EditorConsoleAppender extends AbstractAppender {
+
+    private CircularBuffer<ConsoleLogEvent> circularBuffer;
+    private static final int BUFFER_SIZE = 1000;
+
+    /**
+     * Creates an instance of EditorConsoleAppender.
+     *
+     * @param name             appender name
+     * @param filter           null if not specified
+     * @param layout           pattern of log messages
+     * @param ignoreExceptions default is true
+     *                         <p>
+     *                         Called by {@link #createAppender(String, Filter, Layout, String, String)}
+     */
+    private EditorConsoleAppender(final String name, final int bufferSize, final Filter filter,
+                                  final Layout<? extends Serializable> layout, final boolean ignoreExceptions) {
+        super(name, filter, layout, ignoreExceptions);
+
```

Comment:
Remove the unnecessary new line

Reply:
Fixed in commit [#11d18ac2bd9d5463a9b2562066d005e5472a7ca0](https://github.com/wso2/carbon-analytics/pull/1259/commits/11d18ac2bd9d5463a9b2562066d005e5472a7ca0)